lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -81,21 +77,14 @@
 	 *     Public API
 	 *************************/
 
-	public Http(AblyRest ably, ClientOptions options) {
-		this.ably = ably;
+	public Http(ClientOptions options, Auth auth) {
+		this.options = options;
+		this.auth = auth;
+		this.activeHost = options.restHost;
 		this.scheme = options.tls ? "https://" : "http://";
 		this.port = Defaults.getPort(options);
 	}
 
-	private String getPrefHost() {
```

Comment:
but `AblyRealtime extends AblyRest`, so I think in the realtime case it is true.